lib/spack/llnl/util/filesystem.py

```diff
@@ -2370,7 +2379,27 @@ def add_library_dependent(self, *dest):
                 new_pth = pathlib.Path(pth).parent
             else:
                 new_pth = pathlib.Path(pth)
-            self._additional_library_dependents.add(new_pth)
+            path_is_in_prefix = new_pth.is_relative_to(self.pkg.prefix)
+            add_lib = False
+            if self.link_install_prefix:
+                # We're creating RPath's post install, accept rpath targets anywhere but warn if
+                # out of install prefix
+                if not path_is_in_prefix:
+                    tty.warn(
+                        f"Generating rpath in {new_pth} which is \
+not rooted in target package ({self.pkg}) prefix"
+                    )
```

Comment:
I'm thinking this should actually be an error (when `link_install_prefix` is true).